File: Back End/notifications/src/main/java/com/theelite/notifications/communication/UserService.java
package com.theelite.notifications.communication;

import com.theelite.notifications.model.User;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface UserService {

    @GET("/")
    Call<User> getUserWithId(@Body User user);

    @GET("/validateUser/{email}/{token}")
    Call<Boolean> validateUser(@Path("email") String email, @Path("token") String token);

    @GET("/familyAccount/{email}")
    Call<String> getFamilyAccount(@Path("email") String email);
}

File: Front End/Android/app/src/main/java/com/theelite/portal/request/UserService.kt
package com.theelite.portal.request

import com.theelite.portal.Objects.User
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.PUT

interface UserService {

    @POST("user/auth")
    fun authenthicateUser(@Body user: User): Call<String>

    @PUT("user/new")
    fun newUser(@Body user: User): Call<String>

    @POST("user/get")
    fun getUserInfo(@Body user: User): Call<User>


}